joern-cli/frontends/gosrc2cpg/src/main/scala/io/joern/gosrc2cpg/astcreation/AstCreatorHelper.scala

```diff
@@ -37,6 +38,18 @@ trait AstCreatorHelper { this: AstCreator =>
 
   }
 
+  protected def addModifier(node: NewNode, name: String): Unit = {
+    // NOTE: In golang, the access modifiers are exported and un-exported.
+    // If the first letter of the node (function, typeDecl, etc) is uppercase, then it is exported.
+    // Else, it is un-exported
+    // The scope of the node is the package it is defined in.
+    if (name(0).isUpper) {
+      diffGraph.addEdge(node, NewModifier().modifierType(ModifierTypes.PUBLIC), EdgeTypes.AST)
```

Comment:
You will find a method inside `NodeBuilder.newModfierNode` make use of it. Avoid adding to diffGraph directly. 

refer this https://github.com/joernio/joern/blob/master/joern-cli/frontends/javasrc2cpg/src/main/scala/io/joern/javasrc2cpg/passes/AstCreator.scala#L596 and its corresponding inner methods to build the TypeDecl Node